WORK-289:

hsuchie4 opened the pull request:
- filename keywords will be added into 'path_keywords' of keywords_data
- user defined keywords will be added into 'user_keywords' of keywords_data
- user defined keywords will automatically be updated in keywords_data
- added the global trigger for keyword update

hsuchie4 committed:
WORK-289:
-

oboreo commented on src/tactic/command/global_search_trigger.py:
```diff
@@ -86,6 +86,12 @@ def execute(my):
+        # Updates for adding changed keywords into asset keywords_data column
+        if sobj.column_exists("keywords_data"):
+            user_keywords = input.get("update_data").get("keywords")
+            if user_keywords not in ['none', None]:
```
why would user_keywords be 'none' at times?, should eliminate that case on the js side

oboreo commented on src/tactic/ui/tools/ingest_wdg.py:
```diff
@@ -1602,26 +1604,24 @@ def execute(my):
-            file_keywords = Common.extract_keywords_from_path(path)
+            # Extracting keywords from filename only because all Assets will 
+            # have the keywords [project_name]/[search_type name]/[Ingest]
+            file_keywords = Common.extract_keywords_from_path(filename)
```
# this is debatable.. Should we extract keywords from the full path if relative_dir is not used for the IngestUploadWdg?

oboreo commented on src/tactic/ui/tools/ingest_wdg.py:
```diff
@@ -1602,26 +1604,24 @@ def execute(my):
-
 
+                data = sobject.get_json_value("keywords_data", {})
+                data['user_keywords'] = keywords
```
rename these without the suffix _keywords

hsuchie4 committed:
WORK-289:
- when user keywords are changed, the keywords_data will be updated
- as well as all of its child asset/collections will be updated
- removing assets from a collection will update the asset's collection keywords_data
- deleting a collection will update its asset's collection keywords_data
- changes on keywords_data column will update "searchable_keywords" column
